pkgs/applications/virtualization/containerd/default.nix

```diff
@@ -14,34 +13,37 @@ stdenv.mkDerivation rec {
     sha256 = "16zn6p1ky3yrgn53z8h9wza53ch91fj47wj5xgz6w4c57j30f66p";
   };
 
+  goPackagePath = "github.com/containerd/containerd";
+  outputs = [ "bin" "out" "man" ];
+
   hardeningDisable = [ "fortify" ];
 
-  buildInputs = [ removeReferencesTo go btrfs-progs ];
+  buildInputs = [ btrfs-progs go-md2man utillinux ];
   buildFlags = "VERSION=v${version}";
 
   BUILDTAGS = []
     ++ optional (btrfs-progs == null) "no_btrfs";
 
-  preConfigure = ''
-    # Extract the source
-    cd "$NIX_BUILD_TOP"
-    mkdir -p "go/src/github.com/containerd"
-    mv "$sourceRoot" "go/src/github.com/containerd/containerd"
-    export GOPATH=$NIX_BUILD_TOP/go:$GOPATH
-'';
-
-  preBuild = ''
-    cd go/src/github.com/containerd/containerd
+  buildPhase = ''
+    cd go/src/${goPackagePath}
     patchShebangs .
+    make binaries
   '';
 
   installPhase = ''
-    mkdir -p $out/bin
-    cp bin/* $out/bin
-  '';
-
-  preFixup = ''
-    find $out -type f -exec remove-references-to -t ${go} '{}' +
+    for b in bin/*; do
+      install -Dm555 $b $bin/$b
+    done
+
+    make man
+    manRoot="$man/share/man"
+    mkdir -p "$manRoot"
+    for manFile in man/*; do
+      manName="$(basename "$manFile")" # "docker-build.1"
+      number="$(echo $manName | rev | cut -d'.' -f1 | rev)"
+      mkdir -p "$manRoot/man$number"
+      gzip -c "$manFile" > "$manRoot/man$number/$manName.gz"
```

Comment:
I think we automatically compress manpages.

Reply:
oh :thinking: then there is some other packages that should change too